src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test.Common/CSharpTestLspServerHelpers.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Composition;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Common
+{
+    internal class CSharpTestLspServerHelpers
+    {
+        public static async Task<TResponse> ExecuteCSharpRequestAsync<TRequest, TResponse>(
+            RazorCodeDocument codeDocument,
+            Uri csharpDocumentUri,
+            ServerCapabilities serverCapabilities,
+            TRequest requestParams,
+            string methodName,
+            CancellationToken cancellationToken) where TRequest : class
+        {
+            var csharpSourceText = codeDocument.GetCSharpSourceText();
+            var files = new List<(Uri, SourceText)>
+            {
+                (csharpDocumentUri, csharpSourceText)
+            };
+
+            var exportProvider = RoslynTestCompositions.Roslyn.ExportProviderFactory.CreateExportProvider();
+            using var workspace = CreateCSharpTestWorkspace(files, exportProvider);
+            await using var csharpLspServer = await CreateCSharpLspServerAsync(workspace, exportProvider, serverCapabilities);
```

Comment:
Interesting, so for every request we create a new server? What if we were to have this containing method take in a csharp server OR move the method to the CSharpTestLSPServer type? That way the server capabilities don't have to be passed each time.